### ISSUE TITLE
Hardcoded versions in requirements.txt

### ISSUE THREAD

**Issue by appi147** [sukeesh/Jarvis#352]
There are hard-coded versions in requirements.txt file which are very old. Should we remove hard-coded versions and provide support to latest  versions?
Some people generally install system-wide(sudo) packages and these hard-coded versions mess up some dependencies of another packages.
What do you guys think?

**Comment by pnhofmann** [comment]
Agree! I actually do prefer if requirements are always up-to-date plugins - why disclaim bug fixes and improvements?
With quick tests I did not notice compatibility problems - and don't think those are common with python (?). Compared to requirements for online services like IMDbPY - which stops working if they are too old.

**Comment by pnhofmann** [comment]
Noticed latest _py-translator_ conflicts with old version of _beautifulsoup4_ and _requests_.

So - fix version py-translator of un-fix version of  beautifulsoup4 and requests (and others)?

**Comment by appi147** [comment]
Lets remove hard-coded versions

**Comment by appi147** [comment]
@pnhofmann I am unable to find where `py-translator` is used.

**Comment by pnhofmann** [comment]
Translate module uses googletrans - py-translator is a CLI for googletrans and installs googletrans - but yes, py-translator is actually not necessary.

Noticed that is my mistake; googletrans was broken and when I installed py-translator it worked again - probably just because googletrans was updated. And somehow I commited py-translator.

Anyway: googletrans is currelty broken (again?):

https://github.com/ssut/py-googletrans/issues/94

There is googletrans-temp with a fix - but I don't think using that would be a good idea. So would say, we replace py-translator with googletrans and hope it'll work soon.

**Comment by pnhofmann** [comment]
Noticed: googletrans is in requirements.txt - yes you are right: Installing py-translator is useless.

**Comment by appi147** [comment]
Well, I removed it in #382. Could you review that PR?

**Comment by appi147** [comment]
[https://pypi.org/project/mock/](https://pypi.org/project/mock/)

> mock is now part of the Python standard library, available as unittest.mock in Python 3.3 onwards.
> This package contains a rolling backport of the standard library mock code compatible with Python 2.6 > and up, and 3.3 and up.

Mock hasn't been updated since April 6 , 2016 (2.0.0)

**Comment by appi147** [comment]
pync is related to Mac, don't have access to that, can't test

> A simple Python wrapper around the terminal-notifier command-line tool (version 2.0.0), which allows you to send User Notifications to the Notification Center on Mac OS X 10.10, or higher.

**Comment by appi147** [comment]
bs4 is standard library, could not find anything that will break anything.
[Changelog](https://bazaar.launchpad.net/~leonardr/beautifulsoup/bs4/view/head:/NEWS.txt)

**Comment by appi147** [comment]
instantmusic has not been updated after 2.2

**Comment by appi147** [comment]
ibdmpy is latest now

**Comment by appi147** [comment]
Looks like no problem
